test/parallel/test-repl-use-global.js

```diff
@@ -0,0 +1,61 @@
+'use strict';
+
+// Flags: --expose-internals
+
+const common = require('../common');
+const stream = require('stream');
+const REPL = require('internal/repl');
+const assert = require('assert');
+
+common.globalCheck = false;
+
+test();
+
+function test() {
+
+  runRepl(false, (err, output) => {
+    assert.ifError(err);
+    assert.strictEqual(output, 'undefined');
+
+    runRepl(true, (err, output) => {
+      assert.ifError(err);
+      assert.strictEqual(output, '\'tacos\'');
+    });
+  });
+}
+
+function runRepl(useGlobal, cb) {
+  const inputStream = new stream.PassThrough();
+  const outputStream = new stream.PassThrough();
+
+  const opts = {
+    input: inputStream,
+    output: outputStream,
+    useColors: false,
+    terminal: false,
+    prompt: ''
+  };
+
+  // Don't blindly set useGlobal, because we want to test
+  // the default case as well, where useGlobal is not provided.
+  if (useGlobal)
```

Comment:
Sure. Going back to look at `createInternalRepl`, I see that my comment here doesn't really make sense. You [never get the default](http://www.ecma-international.org/ecma-262/6.0/#sec-getidentifierreference) if you provide any options at all, even an empty object. In that case, this function defaults nothing. Maybe not a huge deal, since this is internal and only ever called once in `node.js`, and the `REPLServer` constructor pretty much [handles defaults](https://github.com/nodejs/node/blob/master/lib/repl.js#L184-L207). But it does seem less than optimal.